packages/at_persistence_secondary_server/lib/src/config/at_config.dart

```diff
@@ -154,13 +152,27 @@ class AtConfig {
       existingData = await get(configKey);
     } on KeyNotFoundException catch (e) {
       logger.finer('Could not fetch data with NEW config-key | ${e.message}');
+    } on Exception catch (e) {
+      logger.finer('Could not fetch data with NEW config-key | $e');
+      rethrow;
     }
     if (existingData == null) {
+      // If data could not be fetched with the new config-key, try fetching the data
+      // using the old config-key and delete the old key from keystore
       try {
-        existingData = await get(olConfigKey);
-        await (persistenceManager.getBox() as LazyBox).delete(olConfigKey);
+        existingData = await get(oldConfigKey);
+        AtData newAtData = AtData()..data = existingData?.data;
+        HiveKeyStoreHelper.getInstance().prepareDataForKeystoreOperation(
+            newAtData,
+            existingAtData: existingData);
+        // store the existing data with the new key
+        await (persistenceManager.getBox() as LazyBox).put(configKey, newAtData);
```

Comment:
Should we append atSign at the end of the config key to make a valid AtKey? For example `private:blocklist@alice`

Reply:
change has been made as part of [929408d](https://github.com/atsign-foundation/at_server/pull/1622/commits/929408d47379b1ac97dc317a40a834e7219e4cff)